src/main/java/com/mongodb/kafka/connect/source/producer/BsonSchemaAndValueProducer.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.kafka.connect.source.producer;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaAndValue;
+
+import org.bson.BsonBinaryWriter;
+import org.bson.BsonDocument;
+import org.bson.BsonValue;
+import org.bson.codecs.BsonValueCodec;
+import org.bson.codecs.Codec;
+import org.bson.codecs.EncoderContext;
+import org.bson.io.BasicOutputBuffer;
+
+import com.mongodb.kafka.connect.source.MongoSourceConfig;
+
+class BsonSchemaAndValueProducer implements SchemaAndValueProducer {
+  private static final Codec<BsonValue> BSON_VALUE_CODEC = new BsonValueCodec();
+
+  @Override
+  public SchemaAndValue create(
+      final MongoSourceConfig config, final BsonDocument changeStreamDocument) {
+    return new SchemaAndValue(Schema.BYTES_SCHEMA, documentToByteBuffer(changeStreamDocument));
+  }
+
+  private byte[] documentToByteBuffer(final BsonDocument document) {
+    BasicOutputBuffer buffer = new BasicOutputBuffer();
+    try (BsonBinaryWriter writer = new BsonBinaryWriter(buffer)) {
+      BSON_VALUE_CODEC.encode(writer, document, EncoderContext.builder().build());
+    }
+    return buffer.toByteArray(); // Use a copied truncated byte[] value
```

Comment:
@jyemin I couldn't determine a more efficient way to get just the written data as a byte array.

Reply:
Does `document` have to be a `BsonDocument`, or could it be a `RawBsonDocument`?  If it can't be raw, this is the most efficient way to do it.

Reply:
I've used RawBsonDocuments where possible (For the source record values) and updated.